src/watchful/client.py

```diff
@@ -1081,6 +1081,49 @@ def export_preview(mode: str = "ftc") -> Optional[Dict]:
     return api("export_preview", mode=mode)
 
 
+def is_utf8(
+    filepath: str,
+    threshold: float = 0.5,
+    is_fast: bool = True,
+) -> Literal[-1, 0, 1]:
+    """
+    This function attempts to detect if the encoding of the file contents of the
+    given filepath is utf-8. It returns -1 if the detected encoding has a
+    confidence lower than the given threshold, otherwise 0 if the encoding is
+    not utf-8 or 1 if it is. This function may need some tweaking for a very
+    large dataset, but should work with the ``is_fast`` argument set to True by
+    default. This function is suitable to be invoked prior to invoking
```

Comment:
Should we add a call to this function as a mandatory step instead of leaving it to the user to know and implement the call?

Reply:
This is a very valid question, I had thought about it but I did not go for it because:
1. although it is improbable for a utf8 encoded file to be detected as non-utf8, it is possible
2. although it is improbable for a non-utf8 encoded file to be detected as utf8, it is possible

If this call is made mandatory, and if case 1 happens (rarely), then users will not be able to upload their datasets. However, we can make it an intentional loose check (i.e. a `warning`) that does not prevent uploading even if the checker detects non-utf8 when the user is super sure that the dataset file is utf8 encoded.

What do you think?

Reply:
I like the idea of emitting a `warning`. In general, my worry is a silent failure that isn't obvious to the user. Whatever we can do to help them debug problems on their own would be my choice.

Reply:
I think this can be a point to start thinking about logging in the SDK, which can help us and users themselves to supporting user issues. For now, I would cater to this with a "warning" message printed to screen; I'll also add a "force" argument so that a user can upload even if the checker detects non-utf8 when the user is super sure that the dataset file is utf8 encoded.